Docs: using `vtctldclient` in all Online DDL -related examples

shlomi-noach opened the pull request:
Followup to https://github.com/vitessio/website/pull/1619

Switching from `vtctl`/`vtctlclient` based examples to `vtctldclient`.

shlomi-noach committed:
audit and control: use vtctldclient. Use examples 307

Signed-off-by: Shlomi Noach <2607934+shlomi-noach@users.noreply.github.com>

shlomi-noach committed:
replace remaining vtctlclient examples with vtctldclient

Signed-off-by: Shlomi Noach <2607934+shlomi-noach@users.noreply.github.com>

mattlord commented on content/en/docs/19.0/user-guides/schema-changes/managed-online-schema-changes.md:
https://deploy-preview-1622--vitess.netlify.app/docs/19.0/user-guides/schema-changes/managed-online-schema-changes/#ddl_strategy

I think this would be:
```
You will set either the `@@ddl_strategy` session variable, to override the `vtgate` `--ddl_strategy` command line flag value, or the `vtctldclient` --ddl-strategy` flag to control your schema migration strategy, and specifically, to enable and configure online DDL
```

Or was the command-line there in reference to `vtctldclient`, where the flag is `--ddl-strategy`?

Here it seems that we're referring to `vtctldclient` so both flags would have a dash: https://deploy-preview-1622--vitess.netlify.app/docs/19.0/user-guides/schema-changes/managed-online-schema-changes/#executing-an-online-ddl-via-vtctldclient

mattlord replied:
It looks like the `-enable-lag-throttler` flag is using a single dash here: https://deploy-preview-1622--vitess.netlify.app/docs/19.0/user-guides/schema-changes/managed-online-schema-changes/#throttling

shlomi-noach replied:
> You will set either the `@@ddl_strategy` session variable, to override the `vtgate` `--ddl_strategy` command line flag value, or the `vtctldclient` --ddl-strategy` flag to control your schema migration strategy, and specifically, to enable and configure online DDL

That's a bit incomplete. I changed to:

You will either set `vtgate` `--ddl_strategy` command line flag value, or will override it with the `@@ddl_strategy` session variable, or use the `vtctldclient` --ddl-strategy` flag to control your schema migration strategy, and specifically, to enable and configure online DDL.

shlomi-noach replied:
> It looks like the -enable-lag-throttler flag is using a single dash here


Oooh that requires a complete rewrite! The flag is deprecated. Rewritten as:

**NOTE** that at this time (and subject to change) the tablet throttler is disabled by default. Enable it via `vtctldclient UpdateThrottlerConfig --enable <keyspace>`. If the tablet throttler is disabled, schema migrations will not throttle on replication lag.

mattlord commented on content/en/docs/19.0/user-guides/schema-changes/ddl-strategies.md:
The flag name is ambiguous here so I'm unsure if it's right or wrong as-is: https://deploy-preview-1622--vitess.netlify.app/docs/19.0/user-guides/schema-changes/ddl-strategies/#specifying-a-ddl-strategy

Is that referring to the vtgate flag or the vtctldclient flag?

mattlord replied:
Looks like a single dash flag here: https://deploy-preview-1622--vitess.netlify.app/docs/19.0/user-guides/schema-changes/ddl-strategies/#setup

mattlord replied:
I would generalize with `ddl strategy` as a way to include `--ddl-strategy` and `--ddl_strategy` and `@@ddl_strategy` everywhere. Including linking locations like here: https://deploy-preview-1622--vitess.netlify.app/docs/19.0/user-guides/schema-changes/audit-and-control/#running-migrations

shlomi-noach replied:
> The flag name is ambiguous here so I'm unsure if it's ...

fixed

shlomi-noach replied:
> Looks like a single dash flag here ...

fixed

shlomi-noach replied:
> I would generalize with ddl strategy as a way to include 

I'm not sure I follow. Can you please explain again?

mattlord replied:
There's now `ddl_strategy` and `ddl-strategy` parameters/flags. `ddl strategy` or `DDL strategy` would be a more general way to cover the setting/topic. It's minor though.

shlomi-noach replied:
Ah, got it! Let me review the doc.

mattlord reviewed (approved):
Some additional notes, as I reviewed all of the pages in the section:
- I think it would be helpful to denote that it's a vttablet flag here: https://deploy-preview-1622--vitess.netlify.app/docs/19.0/user-guides/schema-changes/table-lifecycle/#automated-lifecycle
- The vtctldlcient flags throughout this page look wrong (using underscores): https://deploy-preview-1622--vitess.netlify.app/docs/19.0/user-guides/schema-changes/advanced-usage/

I will approve now so that you can address my comments as you feel is best, copy the final pages to the 18.0 docs, and then merge.

Thanks!

shlomi-noach committed:
rewrite 'Specifying a DDL strategy'. Fix throttler comment.

Signed-off-by: Shlomi Noach <2607934+shlomi-noach@users.noreply.github.com>

shlomi-noach committed:
fix vtctldclient usage

Signed-off-by: Shlomi Noach <2607934+shlomi-noach@users.noreply.github.com>

shlomi-noach commented:
I've addressed all comments except this one, which I did not fully understand:

> I would generalize with ddl strategy as a way to include --ddl-strategy

shlomi-noach committed:
further fixes

Signed-off-by: Shlomi Noach <2607934+shlomi-noach@users.noreply.github.com>

shlomi-noach committed:
Update 18.0 docs

Signed-off-by: Shlomi Noach <2607934+shlomi-noach@users.noreply.github.com>

shlomi-noach commented:
I've completed all changes to the docs, and have copied the changes back to `18.0` docs.

mattlord commented:
> I've completed all changes to the docs, and have copied the changes back to `18.0` docs.

Thanks! 🚢
